RealmWeaver/ModuleWeaver.cs

```diff
@@ -178,11 +178,12 @@ public void Execute()
         var listTypeDefinition = _corLib.MainModule.GetType("System.Collections.Generic.List`1");
         if (listTypeDefinition == null)
         {
-            var collectionsAssembly = AssemblyResolver.Resolve("System.Collections");
-            listTypeDefinition = collectionsAssembly.MainModule.ExportedTypes.FirstOrDefault(x => x.FullName == "System.Collections.Generic.List`1").Resolve();
-            
+            System_IList = _realmAssembly.MainModule.ImportReference(typeof (System.Collections.Generic.List<>));
```

Comment:
I see - it works because we `Resolve()` this `TypeReference` to get its `TypeDefinition` so it doesn't really matter what module imports the reference. Still, I think this could cause subtle bugs later, because this is `IList<>` as referenced from `Realm.dll`, not the woven assembly, and if we ever end up trying to use this reference on the woven assembly things might go wrong.

I strongly recommend replacing this with `ModuleDefinition.ImportReference(typeof(IList<>))` if the latter works.